### PR TITLE
BREAKING CHANGE: Bump version of normalize-url to 8, dependency bumps

### PR DESCRIPTION
**Pull request by jonluca**
Bumps dependency versions, fixes lint errors

There's a breaking change in normalize-url v7.0.3 around url query strings - https://github.com/sindresorhus/normalize-url/pull/158 which I figured should be a major version bump

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,13 +33,17 @@
 		"string"
 	],
 	"dependencies": {
-		"normalize-url": "^7.0.2",
-		"re2": "^1.17.2",
```
Comment (sindresorhus) [inline package.json:37]: ?

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 node_modules
 yarn.lock
+.idea
```
Comment (sindresorhus) [inline .gitignore:3]: Don't make unrelated changes.

Comment (jonluca) [reply .gitignore:3]: This is my IDEs config files, happy to remove

File: package.json (hunk 1 above)
Comment (jonluca) [reply package.json:37]: `re2` isn't used anywhere in the project

**Comment by jonluca** [comment]
@sindresorhus `xo` uses nullish coaelscing. What're your thoughts on this breaking change requiring node 14?

File: package.json (hunk 1 above)
Comment (sindresorhus) [reply package.json:37]: It's required by `url-regex-safe`. Usually, `git blame` is a good way to find out why something exists.

File: .gitignore (hunk 2 above)
Comment (sindresorhus) [reply .gitignore:3]: Yes. It's **your** config. It has nothing to do in a repo. It should be in your own global gitignore.

**Comment by sindresorhus** [comment]
I will require Node.js 16 after merging this.

File: package.json (hunk 1 above)
Comment (jonluca) [reply package.json:37]: Makes sense, will add it back. Requiring a build step for this library feels a bit too much - I might open up an issue for discussion on using an alternate implementation

File: .gitignore (hunk 2 above)
Comment (jonluca) [reply .gitignore:3]: Ok

File: package.json (hunk 1 above)
Comment (sindresorhus) [reply package.json:37]: Yeah, we can switch to https://github.com/sindresorhus/super-regex instead. PR welcome.

**Comment by jonluca** [comment]
Sounds good! I think PR is ready now
